src/App.tsx

```diff
@@ -0,0 +1,135 @@
+import { Component } from 'react';
+import { Header, Modal, RestaurantList, SelectBox } from './components';
+import { GlobalStyle } from './global-style';
+import { CategoryFilter, Restaurant, SortFilter } from './types';
+
+import styled, { ThemeProvider } from 'styled-components';
+
+import data from './data/mockData.json';
+import { db } from './db/restaurants';
+
+import variables from './components/styles/variables';
+
+type Props = {};
+
+type State = {
+  categoryOptions: CategoryFilter[];
+  sortingOptions: SortFilter[];
+  selectedCategoryOption: CategoryFilter;
+  selectedSortingOption: SortFilter;
+  restaurantItems: Restaurant[];
+  isModalOpen: boolean;
+  clickedRestaurant?: Restaurant | null;
+};
+
+class App extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      categoryOptions: ['전체', '한식', '중식', '일식', '아시안', '양식', '기타'],
+      sortingOptions: ['거리순', '이름순'],
+      selectedCategoryOption: '전체',
+      selectedSortingOption: '거리순',
+      restaurantItems: data.items as Restaurant[],
+      isModalOpen: false,
+      clickedRestaurant: null,
+    };
+  }
+
+  componentDidMount() {
+    if (db.isRestaurantItemsExist()) {
+      this.setState({ restaurantItems: db.getRestaurants() });
+    } else {
+      this.setState({ restaurantItems: data.items as Restaurant[] });
+      db.setRestaurants(data.items as Restaurant[]);
+    }
+  }
+
+  render() {
+    return (
+      <>
+        <ThemeProvider theme={{ variables }}>
+          <GlobalStyle />
+          <div className="App">
+            <Header></Header>
```

Comment:
children을 갖지 않는 Component들은 self-closing tag로 작성하시면 좋을거 같네요!
```suggestion
            <Header />
```

Reply:
앗 놓친 부분이네요